rir/src/compiler/opt/loop_invariant.cpp

```diff
@@ -9,19 +9,29 @@ namespace pir {
 
 bool isSafeToHoistLoads(const LoopDetection::Loop& loop) {
     auto noEnvironmentTainting = [](Instruction* i) {
-        if (LdFun::Cast(i))
+        // For this instructions we test later they not change the particular
+        // binding
+        if (StVar::Cast(i) || StVarSuper::Cast(i) || MkEnv::Cast(i))
```

Comment:
you could combine those two checks into one, by just putting the code from below here

Reply:
At least at the moment most of the loops will not be safe to hoist. So I prefer to do a general test first, and then eventually if there are `ldVars` to hoist a particular test regarding only the binding.